src/kraken/core/formatting/exception_traceback.py

```
@@ -0,0 +1,21 @@
+import traceback
+import functools
+
+
+def exception_traceback(stacktrace_depth=2):
+    def exception_handler(func):
+        """Decorator that formats an exception and returns only a part of the stacktrace as specified by the user."""
+
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except Exception:
```

It's probably worth re-raising the exception to prevent swallowing it entirely.

Unfortunately, doing so will mean the exception will reach the interpreter which will generate a full printout ;)

Options are to (a) alter the exception before re-raising it (which feels janky) (b) look at an alternative implementation using something like [`sys.excepthook`](https://docs.python.org/3/library/sys.html#sys.excepthook)

---

@samrogerson I tested an alternative where I assigned sys.excepthook to a formatter function, and raised the exception. This however was similar to the current implementation - where the Exception is raised if the re-raise flag is set. One might argue that this flag is unnecessary, as not using the decorator essentially has the same behaviour. However, the flag makes it more explicit for the user, and with some added fixes on top of this PR coming up (https://github.com/kraken-build/kraken-core/pull/38) I believe this should be fine.